build-images.sh

```diff
@@ -9,50 +9,63 @@ function usage() {
     exit 0
 }
 
-while [ -n "$1" ]; do
-    case $1 in
-    -h | --help) usage ;;
-    -?*)
-        echo "Unknown option '$1'" 1>&2
-        exit 1
+parse_args() {
+  local SHORT=h,d:
+  local LONG=distro:
+
+  local OPTS=$(getopt -o $SHORT -l $LONG --name "$0" -- "$@")
+
+  if [ $? != 0 ] ; then echo "Failed to parse options...exiting." >&2 ; exit 1 ; fi
+
+  eval set -- "$OPTS"
+
+  . settings.sh  # set initial values
```

Comment:
Personally, I do not like this strategy because it does not seem to be very common and I think it can get confusing. I prefer to restrict the usage to the argument interface.

Reply:
I must say that I "disagree" somehow, I agree that it is not common, it is more common using  config file passed as argument, however I think that like this is easier as it is more expansible in number of arguments, maybe for now we can pass from command line and this file?

Reply:
After seeing it closely I agree it is good to have this as a config file, I'll just change the file extension to differentiate it from the other scripts